### ISSUE TITLE
[Bug] comments in python requirements.txt breaks installation process as the pip install takes line by line

### ISSUE THREAD

**Issue by dsaid-xj** [hail-is/hail#10352]
To report a bug, fill in the information below. 
For support and feature requests, please use the discussion forum: 
https://discuss.hail.is/

Please include the full Hail version and as much detail as possible.

-----------------------------------------------------------------------------

Hail commit version: 5852a71e9e23

Error received:
```
amazon-ebs: cd build/deploy; python3 setup.py -q sdist bdist_wheel
amazon-ebs: sed '/^pyspark/d' python/requirements.txt \| xargs python3 -m pip install -U
==> amazon-ebs: ERROR: Invalid requirement: '#'
==> amazon-ebs: make: *** [install-on-cluster] Error 123
```

**Comment by shy218** [comment]
Yes, I receive the same bug today, when I follow the instruction: https://hail.is/docs/0.2/install/other-cluster.html

**Comment by tpoterba** [comment]
I don't think this is actually a bug -- I think this line doesn't support valid requirements.txt files with comments:
```
sed '/^pyspark/d' python/requirements.txt \| xargs python3 -m pip install -U
```

Instead, let's try:

```
cat python/requirements.txt | sed '/^pyspark/d' | grep -v "^#"
```

**Comment by tpoterba** [comment]
oh, this is code in our makefile!

**Comment by tpoterba** [comment]
will fix.

**Comment by tpoterba** [comment]
#10371 should fix

**Comment by tpoterba** [comment]
The fix is merged, please reopen if this is still a problem.

**Comment by mhebrard** [comment]
Still happening on 0.2.65

**Comment by tpoterba** [comment]
The fix landed after the tagged 0.2.65 release. Can you share the git commit hash that's failing?
